src/core/languages/cpp.ts

```
@@ -0,0 +1,18 @@
+import type { LanguageConfig } from './index.js';
+
+export const CppConfig: LanguageConfig = {
+  name: 'C++',
+  extensions: ['.cpp', '.cc', '.cxx', '.hpp', '.h', '.hxx'],
+
+  loadGrammar: async () => {
+    const CppLanguage = await import('tree-sitter-cpp');
+    return CppLanguage.default;
+  },
+
+  extractSymbols: (tree: Parser.Tree, sourceCode: string): Symbol[] => {
```

ESLint

[error] 'tree' is defined but never used. Allowed unused args must match /^_/u.

(@typescript-eslint/no-unused-vars)